src/Storages/StorageMergeTree.cpp

```diff
@@ -1697,5 +1697,33 @@ void StorageMergeTree::populateCommittedSNFromParts()
         LOG_DEBUG(log, "Loading idempotent key={}", *keys_iter->second);
     }
 }
+
+Int64 StorageMergeTree::maxPartsCommittedSN() const
+{
+    auto data_parts = getDataPartsVector();
+    auto committed = committedSN();
+
+    SequenceRanges sequence_ranges;
```

Comment:
If the table does not has any data inserted. and sequence_ranges will be empty. What will happen?

Reply:
In this case:
```
    if (sequence_ranges.empty())
    {
        /// { missing_ranges, next_expecting_sn, max_expecting_sn }
        return {{}, committed + 1, committed + 1};
    }
```